src/cli/argv_parser.ts

```diff
@@ -102,6 +102,12 @@ const ArgvParser = {
   parse(argv: string[]): IParsedArgv {
     const program = new Command(path.basename(argv[1]))
 
+    // We want to be in full control over when the process exits.
+    // This is important for other tools using Cucumber as a library,
+    // such as https://github.com/cucumber/cucumber-electron
+    // The exception is --help, --version (and -v) which will still exit the process immediately
+    program.exitOverride()
```

Comment:
This seems to break the fail fast principle. Cucumber should stop when it can't make sense of the input. This prevents unexpected behavior and simplifies debugging.

Rather then using the CLI I would expect tools that use cucumber-js in process to use the same underlying api that the terminal uses. If that api does not exist, consider making it available.

Alternatively Cucumber JVM has two main methods. A real one and one that returns the exit code but doesn't exit. It's still a bit hacky but does avoid the problems mentioned above.